tasks/sep/004-fpac-upgrade/README.md

```diff
@@ -0,0 +1,198 @@
+
+5. Generate the `input.json` with `just generate-input`
+
+6. Prepare the periphery infrastructure.
+
+   - Point the `op-challenger` and `dispute-mon` towards the new `DisputeGameFactory` contract proxy, deployed in step 3.
+   - Point the `op-proposer` towards the new `DisputeGameFactory` contract proxy, deployed in step 3.
+
+7. Collect signatures for the `OptimismPortal` proxy upgrade.
+
+8. Execute the `OptimismPortal` proxy upgrade, enabling dispute-game based withdrawals on the network.
+
+## Approving the Transaction
+
+### 1. Update repo and move to the appropriate folder for this rehearsal task:
+
+```
+cd superchain-ops
+git pull
+just install
+cd tasks/sep/004-fpac-upgrade
+```
+
+### 2. Setup Ledger
+
+Your Ledger needs to be connected and unlocked. The Ethereum application needs to be opened on Ledger with the message "Application is ready".
+
+### 3. Simulate and validate the transaction
+
+Make sure your ledger is still unlocked and run the following.
+
+Remember that by default `just` is running with the address derived from `/0` (first nonce). If you wish to use a different account, run
+`just simulate [X]`, where X is the derivation path of the address that you want to use.
+
+You will see a "Simulation link" from the output.
+
+Paste this URL in your browser. A prompt may ask you to choose a
+project, any project will do. You can create one if necessary.
+
+Click "Simulate Transaction".
+
+We will be performing 3 validations and extract the domain hash and
+message hash to approve on your Ledger:
+
+1. Validate integrity of the simulation.
+2. Validate correctness of the state diff.
+3. Validate and extract domain hash and message hash to approve.
+
+#### 3.1. Validate integrity of the simulation.
+
+Make sure you are on the "Overview" tab of the tenderly simulation, to
+validate integrity of the simulation, we need to check the following:
+
+1. "Network": Check the network is Sepolia (`11155111`).
+2. "Timestamp": Check the simulation is performed on a block with a recent timestamp (i.e. close to when you run the script).
+3. "Sender": Check the address shown is your signer account. If not, you will need to determine which “number” it is in the list of
+   addresses on your ledger. By default the script will assume the derivation path is m/44'/60'/0'/0/0. By calling the script with
+   `just simulate 1` it will derive the address using `m/44'/60'/1'/0/0` instead.
+
+#### 3.2. Validate correctness of the state diff and events.
+
+1. Simulate the transaction with `SIMULATE_WITHOUT_LEDGER=1 just simulate`
+1. Copy the Tenderly simulation URL and navigate to it in your browser.
+1. Double check that the state diff affects the correct storage slots.
+
+![](./images/tenderly-state-diff.png)
+
+**Expected Modified Slots**
+
+- `OptimismPortal` Proxy
+  - `0x0000000000000000000000000000000000000000000000000000000000000037` - Sets the `SystemConfig` proxy address.
```

Comment:
The currently manual validation with Tenderly is definitely tedious, especially because for some reason they don't always decode things, so there is room for improvement here. @maurelian started on a [StateDiffChecker script](https://github.com/ethereum-optimism/superchain-ops/blob/87cc55e66e01eb13b761b5e15600f80c75966700/script/LibStateDiffChecker.sol#L8), we can probably reuse some of that to do things like this

Created https://github.com/ethereum-optimism/superchain-ops/issues/99 to track